control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -407,3 +494,45 @@ impl ResourceReplicas for OperationGuardArc<VolumeSpec> {
         Ok(())
     }
 }
+
+#[async_trait::async_trait]
+impl ResourceShutdownOperations for OperationGuardArc<VolumeSpec> {
+    type RemoveShutdownTargets = DestroyShutdownTargets;
+    type Shutdown = ();
+
+    async fn shutdown(
+        &mut self,
+        _registry: &Registry,
+        _request: &Self::Shutdown,
+    ) -> Result<(), SvcError> {
+        // not applicable for volume
+        unimplemented!()
+    }
+
+    async fn remove_shutdown_targets(
+        &mut self,
+        registry: &Registry,
+        request: &Self::RemoveShutdownTargets,
+    ) -> Result<(), SvcError> {
+        let shutdown_nexuses = registry
+            .specs()
+            .get_volume_shutdown_nexuses_guards(&request.uuid)
+            .await?;
+        let mut all_targets_destroyed = true;
+        for mut nexus in shutdown_nexuses {
+            let nexus_spec = nexus.lock().clone();
+            let destroy_req = DestroyNexus::from(&nexus_spec).with_disown(&request.uuid);
+            // the targets which weren't destroyed keep them in deleting state so that reconcilers
+            // can pick them up.
+            if nexus.destroy(registry, &destroy_req).await.is_err() {
+                // flag that all targets weren't destroyed
+                all_targets_destroyed = false;
+            }
+        }
+        if all_targets_destroyed {
+            Ok(())
+        } else {
+            Err(SvcError::Deleting {})
+        }
```

Comment:
I think here we might simply return success as long as the specs are marked as deleting? We'd need to ensure we updated etcd correctly.

Reply:
The specs would be marked as deleting as that is the actual destroy call that takes care of it. I have added this error in case cluster agent needs to do something with the error. Otherwise we can simply return success.

Reply:
I think cluster agent is happy to have us retry on our reconcilers IIRC @mtzaurus ?
But to be able to do that I think we need to check what kind of errors we got, example if we could not reach etcd then we can't ignore the error because if we restart, we'll "forget" that we tried to do this and won't reconcile.

Reply:
Done.